uint16 tool_mask
---
bool uttered